azkaban-common/src/main/java/azkaban/imagemgmt/daos/HPFlowDaoImpl.java

```diff
@@ -0,0 +1,167 @@
+package azkaban.imagemgmt.daos;
+
+import azkaban.db.DatabaseOperator;
+import azkaban.db.SQLTransaction;
+import azkaban.executor.ExecutableFlow;
+import azkaban.imagemgmt.exception.ErrorCode;
+import azkaban.imagemgmt.exception.ImageMgmtDaoException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.apache.log4j.Logger;
+
+import static azkaban.imagemgmt.utils.ErroCodeConstants.*;
+
+
+/**
+ * Dao implementation for managing high priority flows
+ */
+@Singleton
+public class HPFlowDaoImpl implements HPFlowDao {
+  private static final Logger log = Logger.getLogger(HPFlowDaoImpl.class);
+
+  private final DatabaseOperator databaseOperator;
+
+  private static String INSERT_HP_FLOW_OWNER =
+      "INSERT into hp_flow_owners (name, created_on, created_by, "
+      + "modified_on, modified_by ) "
+      + "values (?, ?, ?, ?, ?)";
+
+  private static String INSERT_HP_FLOWS =
+      "INSERT into hp_flows (flow_id, created_on, created_by) "
```

Comment:
What's an example of `flow_id`? If I remember correctly Azkaban doesn't generate numeric flow ids like it does for executions. It uses the flow name as the id but the same name can be used in different projects. How are we resolving name conflicts?

Reply:
Good point, I will add a comment. The flow id here is,
project_name.flow_name